### ISSUE TITLE
SYSTEM: Test and make robust against DNS server loss

### ISSUE THREAD

**Issue by John-Holt-Tessella** [ISISComputingGroup/IBEX#4120]
As RIKENFE instrument scientist if the DNS server disappears I want my system to carry on. 

We need a test for this and to make it robust

**Comment by FreddieAkeroyd** [comment]
Do we mean loss of DNS as a result of a network failure, or just loss of DNS?

**Comment by ChrisM-S** [comment]
Generally, we have avoided use of DNS or other server supplied names (e.g WINS)  in VIs and IOCs for network resources on the instruments and just used fixed or reserved IP addresses.  I'm not sure how careful we have been about this recently (names are attractive) but this probably explains the comment in the original message about the instrument communication continuing.   There are ways of addressing by names which are stored locally (and hence don't need a remote lookup), but needing to use name lookup on a server remote from the instrument potentially prevent the instrument from running in the event of a partial network failure or DNS failure (which could be otherwise mitigated).

As this has happened and on a cycle day could have lost beam time , we should perhaps review our methodology here?

**Comment by KathrynBaker** [comment]
We aren’t using names as a general solution, we are using IP addresses.

There is one exception of certain Lakeshores where there is a firmware bug in the Lakeshore software (yes, it has been logged in their systems) where the device is unable to show us the IP address reliably (it lies, it gives us bad IP address information, the name is therefore safer to use).

Whatever happened today was probably not limited to the DNS, but was a major issue with the network, and some items were resilient and recovered, and some didn’t.

It is also worth remembering that what got lost on RIKENFE was only a partial aspect of the system, and was serial comms via MOXAs.

Whilst I agree we need to make the system more robust given the number of issues we have had today, I’m not sure we can define what that test is until we are sure what caused the issues. There is a general consensus that it is related to the network and that there was a loss of a DNS server, but that doesn’t explain the issues on the South Side Muons Galil interactions, which would seem to be related.

**Comment by FreddieAkeroyd** [comment]
The problem today was not caused by loss of DNS (we also have redundancy on DNS that would have kicked in). The R55 DHCP server failed and was offline for long enough that DHCP lease times expired on items (such as cabin PCs) that do not have extended lease times. MOXA boxes are OK unless you reboot them when DHCP is not running. We need to check if we have all equipment required on an extended DHCP lease time, maybe we are missing some. If they don't support the extended lease time we may need to use static IP addresses.

**Comment by ChrisM-S** [comment]
Looking at reservations, they appear to all have class `InstrumentDHCPUC` which makes me think any reservations will already be on the 7 day lease - if confirmed, I think we should also reserve the named/critical cabin PCs.  As they are essentially fixed slots, it makes little sense for them to jiggle around on a 4 hour lease for no good reason!

**Comment by Tom-Willemsen** [comment]
Also consider whether we want things like `NDS` chopper machines on a longer lease - this caused an issue with the MAPS digital drive fermi.

**Comment by FreddieAkeroyd** [comment]
Do you need to set the DHCP classid on each client itself  e.g. "ipconfig /setclassid" or can you configure it on the DHCP server reservation itself? Would be useful if we could do it on the server as that would make it applicable in more instances.

**Comment by ChrisM-S** [comment]
It looks to me that it might already be there, but I will check with the one who knows.

**Comment by ChrisM-S** [comment]
Just to clarify after testing on a MOXA (could have been something else dumb), the "InstrumentDHCPUP" class showing on the reservations in R55 and R80 does not seem to make the DHCP poll time for reservations half of 7 days independent of the client.  The MOXA on the reservation contacted the server after two hours to keep it's reservation - so it was clearly being given the poll period of the default lease.  The most likely problem kit is embedded PCs, and ancilliary PCs like Tritons which will re-query DHCP quickly but have not been setup with the 7 day lease class.  A hitlist might be worth building of machines like this to fix. Starting with @Tom-Willemsen's  suggestion of the chopper servers (for completeness).

| machines | number | fixed with 7 day class? |
|----------|---|------|
NDS chopper servers|10-15? |YES (static addresses) |
Tritons | 5 | NO|
Other monitoring PCs | ??| NO|
Other chopper PCs e.g. Fermi | ? | NO
What else | ? | ?

**Comment by FreddieAkeroyd** [comment]
Hi @ChrisM-S  can you not run    ipconfig /setclassid   on the Triton and other PCs?

**Comment by ChrisM-S** [comment]
Yes I think so, just wondering how many machines could be fixed and where they might be.  I think the MOXAs are embedded Linux so is it likely that the default for similar DHCPed Linux boxes would keep their lease - or is it technically a (helpful) fault of the MOXAs?

**Comment by ChrisM-S** [comment]
Just for the record, while I had MOXAs on test I did check the behaviour of a MOXA explicitly for DHCP loss when it went past it's 4 hour lease, for a day or two.  The address was maintained and remained functional (even when having a network cable connected/disconnected).  When reconnected to a network (with no DHCP server) it communicated on this held address.

**Comment by KathrynBaker** [comment]
There may be more that can be done, which should be considered in the other network discussions, but this incident was now a long time ago, and these are rare occurrences. Closing in favour of new tickets when there is an oportunity to deal with them, please reopen if this is not acceptable.
